ElectronClient/gui/MainScreen.jsx

```diff
@@ -75,6 +75,13 @@ class MainScreenComponent extends React.Component {
 		}
 	}
 
+	toggleTheme() {
+		this.props.dispatch({
+			type: 'THEME_TOGGLE',
+		});
```

Comment:
It looks like the UI is sending this event.  If so, there should be feature tests for this, to check the app is updated correctly after this event is processed.

Reply:
Thanks. I'll look into it 👍

Reply:
`THEME_TOGGLE` event changes the current state and also set updated state to settings `isNightMode` value, I tried running a build test for the same like:
```javascript
testApp.dispatch({
   type: 'THEME_TOGGLE',
});
await testApp.wait();

const state = testApp.store().getState();
expect(state.settings.isNightMode).toBe(true);
```
but that didn't work, I also tried for `THEME_TOGGLE_SET` still testes failed.
Then, I looked for other similar events like `NOTE_VISIBLE_PANES_TOGGLE` I ran tests for this events still the tests were failed so I don't think build tests are required for such events...
Let me know if any other build tests are required.

Reply:
Unfortunately I don't have time to look into this now, but a test of the type you describe is what I expect to work.

Do we need to wait some time for settings to be updated, or are they updated immediately?  My other guess is perhaps the `Settings` functionality is disabled in the feature tests?  But I'm not sure without looking closer.

Reply:
No, the Night mode button changes the setting spontaneously, As I said earlier this button is just same as Toggle Notes button which controls visibility of Notes pane so I simply had followed the exact implementation of Toggle Notes button for Night mode button who just toggle theme from preferred Light theme to preferred dark theme.

Reply:
Ok, will have a look at it when I can.

Reply:
All good, apart from the generic setting message, the theme messages are handled only within the Electron client so the feature test cannot help here.